ee/server/service/teams.go

```diff
@@ -1523,6 +1518,9 @@ func unmarshalWithGlobalDefaults(b *json.RawMessage) (fleet.Features, error) {
 func (svc *Service) updateTeamMDMDiskEncryption(ctx context.Context, tm *fleet.Team, enable *bool) error {
 	var didUpdate, didUpdateMacOSDiskEncryption bool
 	if enable != nil {
+		if svc.config.Server.PrivateKey == "" {
```

Comment:
You should be able to disable even if `svc.config.Server.PrivateKey == ""`, right?